### ISSUE TITLE
Diffusion Policy: "Blind" trajectory following at 5k steps, stationary jittering at 30k steps (Overfitting or Image Mismatch or less  step?)

### ISSUE THREAD

**Issue by flyingTeng** [huggingface/lerobot#2761]
I am training a Diffusion Policy on a small dataset (100 episodes, ~40k frames) and severe "Chaotic/Erratic" movement at 30k steps (Image Transform Mismatch?) 
At ~5,000 steps: The robot executes a smooth trajectory, but it seems "visually blind". It follows a fixed average trajectory regardless of where the object is placed or if the object is present. It acts based on priors rather than visual conditioning
After 5,000 steps (up to 30,000 steps): The performance degrades instead of improving. The robot stops executing the trajectory and starts jittering/squirming at the initial position with very small amplitude. It seems like the model has collapsed or overfitted heavily
`Dataset: 100 episodes

Backbone: ResNet18

Obs Horizon: 2

Pred Horizon: 16

Action Steps (Chunking): 8

Batch Size: 16

Total Steps: 30,000

Image Transform: crop_shape=[96, 96], crop_is_random=true`
Is 30k steps causing gradient explosion on small datasets? Could the "chaotic movement" also be a sign of overfitting/gradient explosion because 30k steps is too long or less for 100 episodes?

**Comment by flyingTeng** [comment]
<img width="2492" height="1051" alt="Image" src="https://github.com/user-attachments/assets/7375ba23-20f0-41a4-a537-ff556c8621cd" />

**Comment by flyingTeng** [comment]
Action Space & Normalization Strategy:The robot has an 8-dimensional action space consisting of 7 joint angles and 1 gripper state. I am applying a custom manual normalization strategy:Arm Joints (0-6): The joint angles (in radians) are normalized by dividing by $\pi$ (mapping the range $[-\pi, \pi]$ to approx. $[-1, 1]$).Gripper (7): The gripper state is normalized to the range $[0, 1]$.

**Comment by flyingTeng** [comment]
Comparison with ACT: I also trained an ACT policy on the exact same dataset (100 episodes). The ACT policy performs successfully at 10k steps. This confirms that the dataset quality and robot hardware integration are correct.

**Comment by sotanakamura** [comment]
Please check if the crop_shape value is appropriate. In the paper, it is set to 90% of the original image.

**Comment by flyingTeng** [comment]
> 请检查 crop_shape 的值是否合适。论文中将其设置为原始图像的 90%。
I trained for 100k steps and found that the vision layer isn't working at all. It really looks like the reason is that my crop_shape was set too small ，thank you

**Comment by brysonjones** [comment]
Hi @flyingTeng 

I'm working on integrating a multitask version of diffusion policy in this PR: https://github.com/huggingface/lerobot/pull/2545

It's close to merging, and I would love to hear your experience with it!

**Comment by brysonjones** [comment]
Following up on this, https://github.com/huggingface/lerobot/pull/2545 was merged in.

If you try this model, it would be great to know how it goes for you!
